### ISSUE TITLE
MA0003 should not be triggered for well-known collection methods

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#588]
### Version of the Meziantou.Analyzer NuGet package

2.0.74

### Rule Identifier

MA0003

### Target Framework

net8.0

### C# Language version

C# 12

### Description

MA0003 is triggered for collection methods like `Add`/`Contains`/`Remove`/`(Last)IndexOf`.

### Reproduction Steps

Minimal code:

```c#
IReadOnlySet<string?> set = ...;

_ = set.Contains(null);
```
